pkg/cli/debug/debug.go

```diff
@@ -375,11 +376,52 @@ func (o *DebugOptions) RunDebug() error {
 	if err != nil {
 		klog.V(4).Infof("Unable to get exact template, but continuing with fallback: %v", err)
 	}
+	var debugNodeNS *corev1.Namespace
+	generateName := names.SimpleNameGenerator.GenerateName("openshift-debug-node-")
+	// create namespace only if debugging a node
+	if o.IsNode && len(o.ToNamespace) == 0 && len(o.Namespace) == 0 {
+		if !o.Attach.Quiet {
+			fmt.Fprintf(o.Attach.ErrOut, "\nCreating debug namespace/%s ...\n", generateName)
+		}
+		debugNodeNS, err = o.CoreClient.Namespaces().Create(context.TODO(), &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: generateName,
+				Labels: map[string]string{
+					"openshift.io/run-level": "0",
+				},
+				Annotations: map[string]string{
+					"openshift.io/node-selector": "",
+				},
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	defer func() {
+		if debugNodeNS != nil {
+			if !o.Attach.Quiet {
+				fmt.Fprintf(o.Attach.ErrOut, "Removing debug namespace/%s ...\n", generateName)
+			}
+
+			if err := o.CoreClient.Namespaces().Delete(context.TODO(), debugNodeNS.Name, metav1.DeleteOptions{}); err != nil {
+				fmt.Printf("%v", err)
```

Comment:
I'd print that to `o.ErrOut` just like above.

Reply:
updated